app/controllers/api/base_controller/generic_objects_mixin.rb

```diff
@@ -0,0 +1,40 @@
+module Api
```

Comment:
I wonder if this should go in a new folder: app/controllers/api/mixins/ 
and dropping the BaseController in the module hierarchy, it would just be a submodule of the
class including the mixin.

Reply:
@abellotti I explained a bit more in the PR description, but I have been looking into this (which is why it's still WIP) ...however, without the BaseController module hierarchy and with it in `app/controllers/api/mixins`, it has been creating sporadic test failures that I have been trying to figure out.

Reply:
It seems to have to do with the way things are loaded, but haven't figured out the solution just yet.